src/compiler/crystal/tools/doc/method.cr

```diff
@@ -57,15 +57,12 @@ class Crystal::Doc::Method
   private def compute_doc_info : DocInfo?
     def_doc = @def.doc
     if def_doc
-      has_inherit = def_doc =~ /^\s*:inherit:\s*$/m
-      if has_inherit
-        ancestor_info = self.ancestor_doc_info
-        if ancestor_info
-          def_doc = def_doc.gsub(/^[ \t]*:inherit:[ \t]*$/m, ancestor_info.doc.not_nil!)
-          return DocInfo.new(def_doc, nil)
-        end
+      ancestor_doc_info = nil
+      # TODO: warn about `:inherit:` not finding an ancestor
+      inherit_def_doc = def_doc.gsub(/^[ \t]*:inherit:[ \t]*$/m) { (ancestor_doc_info ||= self.ancestor_doc_info).try(&.doc) || break }
 
-        # TODO: warn about `:inherit:` not finding an ancestor
+      if inherit_def_doc && !inherit_def_doc.same?(def_doc)
```

Comment:
`inherit_def_doc` is of type `String`, I think `if inherit_def_doc` doesn't do anything

Reply:
The type is `String?` because we're breaking from the block when no ancestor doc_info is found.

Reply:
I see. Wow 😵‍💫
Well, the code is definitely way more confusing than before.